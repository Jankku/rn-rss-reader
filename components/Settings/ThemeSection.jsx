import { useContext } from 'react';
import { View, Switch, Text } from 'react-native';
import { useTheme } from '@react-navigation/native';
import { Ionicons } from '@expo/vector-icons';
import SettingsSectionHeader from './SettingsSectionHeader';
import SettingsSectionItem from './SettingsSectionItem';
import { ThemeContext } from '../../context/ThemeContext';

function ThemeSection() {
  const { colors } = useTheme();
  const { isDark, toggleTheme } = useContext(ThemeContext);

  return (
    <>
      <SettingsSectionHeader title={'Theme'} />
      <SettingsSectionItem>
        <Ionicons name={'sunny'} size={24} color={colors.text} style={{ paddingEnd: 16 }} />
        <View style={{ alignItems: 'baseline' }}>
          <Text style={{ color: colors.text }}>Dark mode {isDark ? 'enabled' : 'disabled'}</Text>
          <Switch value={isDark} onValueChange={toggleTheme} thumbColor={colors.primary} />
        </View>
      </SettingsSectionItem>
    </>
  );
}

export default ThemeSection;
