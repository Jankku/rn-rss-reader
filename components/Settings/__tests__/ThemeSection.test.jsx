import ThemeSection from '../ThemeSection';
import { render } from '@testing-library/react-native';

describe('Test ThemeSection', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(<ThemeSection />);
    expect(toJSON()).toMatchSnapshot();
  });
});
