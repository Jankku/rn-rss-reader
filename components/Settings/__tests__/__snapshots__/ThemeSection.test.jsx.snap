// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ThemeSection Should match snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "fontSize": 22,
        "fontWeight": "700",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    Theme
  </Text>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Dark mode
        </Text>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="rgb(0, 122, 255)"
          value={false}
        />
      </View>
    </View>
  </View>,
]
`;
